src/jinja2/bccache.py

```diff
@@ -79,7 +79,9 @@ def load_bytecode(self, f: t.BinaryIO) -> None:
             self.reset()
             return
 
-    def write_bytecode(self, f: t.BinaryIO) -> None:
+    def write_bytecode(
+        self, f: t.Union[t.BinaryIO, tempfile._TemporaryFileWrapper]
```

Comment:
I'm not thrilled about this, but I'm not seeing a better way.  Seems like an oversight that tempfile._TemporaryFileWrapper isn't considered BinaryIO.

Reply:
Nevermind, changed to `IO[bytes]` which seems more kosher.